storage/atsd/atsd.go

```diff
@@ -0,0 +1,400 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package atsd
+
+import (
+	"flag"
+	"fmt"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	info "github.com/google/cadvisor/info/v1"
+	"github.com/google/cadvisor/manager"
+	"github.com/google/cadvisor/storage"
+
+	"github.com/BurntSushi/toml"
+	atsdHttp "github.com/axibase/atsd-api-go/http"
+	atsdStorageDriver "github.com/axibase/atsd-storage-driver/storage"
+	"os"
+	"strconv"
+)
+
+const (
+	startDelay           = 15 * time.Second       //waiting to store enough data for all entities before send
+	timestampPeriodError = 250 * time.Millisecond //time error accumulated during housekeeping
```

Comment:
Time duration between two AddStats calls can be less than housekeeping interval. If we setup our storage driver with a sampling interval = housekeeping interval it will send series to ATSD with reduced frequency. To fix this we add max possible error.